src/core/ready.js

```diff
@@ -49,11 +47,36 @@ jQuery.extend( {
 			return;
 		}
 
-		// If there are functions bound, to execute
-		readyList.resolveWith( document, [ jQuery ] );
+		whenReady = function( fn ) {
+			readyCallbacks.push( fn );
+
+			if ( !readyFiring ) {
+				readyFiring = true;
+
+				// Prevent errors from freezing future callback execution (gh-1823)
+				try {
+					while ( readyCallbacks.length ) {
+						fn = readyCallbacks.shift();
+						if ( jQuery.isFunction( fn ) ) {
+
+							// Prefer sync with no try/catch here
+							// Backwards-compatible, maintain execution order
+							fn.call( document, jQuery );
+						}
+					}
+				} finally {
+					readyFiring = false;
```

Comment:
It would certainly be more Promise-compatible to have

```js
jQuery.ready.then( throws );
jQuery.ready.then( shouldRun );
```
work without waiting for a subsequent `jQuery.ready.then` invocation. And fixes don't get much simpler than `setTimeout( whenReady )`.